app/controllers/application_controller.rb

```diff
@@ -10,4 +11,21 @@ class ApplicationController < ActionController::Base
   def authorize_admin!
     redirect_to root_path, alert: "Not authorized" unless current_user.admin?
   end
+
+  # Auth check for Blazer reporting
+  def require_organization
+    redirect_to root_path unless current_user && current_organization
+  end
+
+  def set_current_user
+    User.current = current_user
+  end
```

Comment:
Im curious why we need this, and we cant just use the `current_user` helper from devise directly.

Reply:
The blazer `data_sources` method that's being overridden doesn't have access to controller-level variables. `current_user` isn't within it's scope.